frontend/appflowy_flutter/lib/core/helpers/url_launcher.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/services.dart';
+
+import 'package:appflowy_backend/log.dart';
+import 'package:url_launcher/url_launcher.dart' as launcher;
+
+typedef OnFailureCallback = void Function(Uri uri);
+
+Future<bool> afLaunchUrl(
+  Uri uri, {
+  OnFailureCallback? onFailure,
+  launcher.LaunchMode mode = launcher.LaunchMode.platformDefault,
+  String? webOnlyWindowName,
+}) async {
+  try {
+    return await launcher.launchUrl(
+      uri,
+      mode: mode,
+      webOnlyWindowName: webOnlyWindowName,
+    );
+  } on PlatformException catch (e) {
```

Comment:
If the `onFailure` callback is null, consider displaying a toast message to inform users of the error.